### ISSUE TITLE
License

### ISSUE THREAD

**Issue by butterflyhug** [pithu/js-joda#10]
@pithu, @phueper -- I think it would be very helpful for this project if the two of you could agree on a license for this code and officially add it to the repo.

You've gotten a little attention recently in discussion at https://github.com/moment/moment/issues/1754#issuecomment-169142952, and I suspect you might pick up another contributor or two from that discussion if your licensing intentions were a bit clearer.  (I might be interested to contribute myself at some point, but I'm pretty swamped for at least the next month or two.)

My personal bias is that the original Joda Time got this right when they picked the Apache License -- but as the two folks who have submitted code, obviously your preferences reign supreme.

Thanks for working on this library!

**Comment by phueper** [comment]
Hi @butterflyhug, 

Thanks for your input... We know that we need to decide on the license, but unfortunately @pithu will not be available for a few weeks so this decision will have to wait some more until he returns...

**Comment by pithu** [comment]
@phueper thx to my smartphone i am still available for discussion, especially if its an interesting one.

hello @butterflyhug,
Thanks for your input. Since we are still quite far away from "production ready code", the license question is not yet central for us. The reason why the license question is still open, is quite simple: I am not an expert for (and not a friend of) license questions.  Our project just copies the java.time API from the openjdk to javascript. The java.time API is under GNU GPL+linking exception, as far as i understand, we have to stick to GPL (copy left). But we are still investigating in these issue and we are open for expert tips and for proposals.

*One word to the goal and roadmap of js-joda: 
js-joda will be a one to one copy of the jdk8 java.time API (as far as possible with javascript) see http://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html for a documentation of that API. We are NOT inventing something new! Its piles of routine piece of work. contribution is welcome!
First milestone is to have LocalDate, LocalDateTime, Duration and Period working. The result of the first milestone will be a fully functional Date/ Time library, converting from and to ISO8601, with UTC and system default timezone. Any further timezone converting and localization stuff is not part of the first milestone and will be saved for later. Me myself will fully concentrate in February/ March on js-joda and i am quite optimistic to get a release candidate done for M1 in that time.*

**Comment by phueper** [comment]
both @pithu and i are pretty indifferent regarding the licensing... we are thankful for any input. 

If we continue to use the JDK implementation as reference the license will probably be GPL + linking exception, we could however also rather use the JSR-310 backport (http://www.threeten.org/threetenbp/)  which is BSD licensed which seems to be more common in the JS community?

**Comment by butterflyhug** [comment]
My personal preference would be not GPL + linking exception for a JavaScript library.  At work I do more development with browser-facing JS than Node.  I don't think our lawyers would let me bundle code with a GPL+linking license into our single minified script file for browsers to download.

Otherwise, I'm also not dogmatic about licensing.  I think BSD would be fine.

Awesome to hear that you have the time to focus on solidifying a good initial milestone feature set in the relatively near term -- thanks again for working on this!

**Comment by phueper** [comment]
@pithu and i discussed this. We are both favoring the BSD license. So we will use the threetenbp as reference. The differences to the JSR-310 in JDK8 seem to be minimal and as far as i see not really relevant for the JS port. 
I will go through the commits and rebase the files in the comming days/weeks. Once that is done we will change the README.md with the license info and then close this issue...

So in summary : we plan to use BSD but it will take some time to rebase and clean up.

**Comment by butterflyhug** [comment]
Sounds great.  Thanks for the update!
